### PR TITLE
Sorting RealmResults is stable.

### PR DESCRIPTION
**Pull request by kneth**
@cmelchior

### CODE REVIEW

**Comment by cmelchior** [comment]
:+1:

File: realm/src/androidTest/java/io/realm/SortTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
+/*
```
Comment (bmunkholm) [inline realm/src/androidTest/java/io/realm/SortTest.java]: All other files that test a class is called CLASSTest.java, so when we test cross class functionality, maybe we should use another naming practice.

File: realm/src/androidTest/java/io/realm/SortTest.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    public void testSortMultiFailures() {
+        RealmResults<AllTypes> allTypes = testRealm.allObjects(AllTypes.class);
+
+        // zero fields specified
+        try {
+            allTypes.sort(new String[]{}, new boolean[]{});
+            fail();
+        } catch (IllegalArgumentException ignored) {}
```
Comment (bmunkholm) [inline realm/src/androidTest/java/io/realm/SortTest.java]: Let's wrap the last }

File: realm/src/androidTest/java/io/realm/SortTest.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+        // number of fields and sorting orders don't match
+        try {
+            allTypes.sort(new String[]{FIELD_STRING}, new boolean[]{RealmResults.SORT_ORDER_ASCENDING, RealmResults.SORT_ORDER_ASCENDING});
```
Comment (bmunkholm) [inline realm/src/androidTest/java/io/realm/SortTest.java]: The ascending/ascending bool array is duplicate all over. Suggest to setup and reuse once for simpler reading.

**Comment by bmunkholm** [comment]
Should also test descending/ascending sort.

File: realm/src/androidTest/java/io/realm/SortTest.java (hunk 1 above)
Comment (kneth) [inline realm/src/androidTest/java/io/realm/SortTest.java]: We do have a `NotificationTest.java` which tests functionality and not a class. I extrapolated a naming practice from this example :-)

**Comment by kneth** [comment]
I have cleaned up the unit tests (increase code reuse) and added extra tests.

@bmunkholm

**Comment by emanuelez** [comment]
:+1:

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**